src/main/java/org/kairosdb/core/http/rest/QueryAutocompleter.java

```diff
@@ -0,0 +1,49 @@
+package org.kairosdb.core.http.rest;
+
+import com.google.common.collect.SetMultimap;
+import org.kairosdb.core.datastore.QueryMetric;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.HashSet;
+import java.util.Set;
+
+public class QueryAutocompleter {
+    private static final Logger logger = LoggerFactory.getLogger(QueryAutocompleter.class);
+
+    public void complete(QueryMetric query) {
+        completeMetricTag(query);
+    }
+
+    private void completeMetricTag(QueryMetric query) {
```

Comment:
We discussed this with @rajatparida86, sorry that it was not written carefully.
The idea was to autocomplete the query only in situation, when 
* the key has wildcard
* there are no other non-wildcard tags, that could be used as a split index.

The first is obvious, so let me elaborate on the second one.
The `metric` has been added with the second priority (after `key`) in split-index tags. That means, that once we will add the metric tag to the query, it would be used on the first place. 
For some queries it could lead to performance degradation, since the cardinality of `metric` tag in general lower, that the cardinality of `entity`, for example. The same could work for `application_id` and other tags. 

So I would propose to check the query, and add the `metric` only in case when it should definitely improve the performance. 

We, unfortunately, have not so much time to deploy fixes, so from my perspective it will be better to be on a safe side.

Reply:
I thought that metric would be added as a last-priority index to not mix up cardinalities.

Reply:
Yes, that was the part that I missed. We discussed this, and will follow the current approach